internal/pkg/agent/control/server/server.go

```diff
@@ -29,27 +30,33 @@ import (
 	"github.com/elastic/elastic-agent/pkg/core/logger"
 )
 
+const (
+	apiStatusTimeout = 45 * time.Second
```

Comment:
Are we sure this is long enough for the entire stream? 

It looks like we don't collect CPU profiles by default (maybe we should be doing that [here](https://github.com/elastic/elastic-agent-client/blob/4b23a52d0ebe15be967185e0c36b56c697052c42/pkg/client/client_v2.go#L658-L665)), but if we did and used the default duration they would each take 30s https://pkg.go.dev/net/http/pprof#Profile

It might be better to use a timeout for each chunk/message but I'm not sure if this is possible.

Reply:
Maybe the 30s default CPU profile duration is the reason we don't collect it by default.

Reply:
[Blocker]
This is a good point. If the diagnostics is set to connect pprof profiles, how could this timeout be configured or is 45s always enough?

Reply:
i'll get rid of it and let client drive the cancellation. seems like better idea